sdk/src/azure/core/az_http_response.c

```diff
@@ -32,6 +32,11 @@ static AZ_NODISCARD bool _az_is_http_whitespace(uint8_t c)
 static AZ_NODISCARD az_result _az_get_digit(az_span* ref_span, uint8_t* save_here)
 {
 
+  if (az_span_size(*ref_span) == 0)
```

Comment:
Per the Embedded C model this should be a precondition check.  Is there a specific scenario which forces us to be more protective at runtime in this situation.

Reply:
well, the value can change during runtime, that's why precondition is not enough for this cases.
We are getting an HTTP response from the server and reading/parsing it.  If the response, at some point becomes incomplete or full of digits only, we are making sure we don't read beyond span size.